db/warehouse/migrate/20250523195117_add_hmis_unit_groups.rb

```diff
@@ -0,0 +1,24 @@
+###
+# Copyright 2016 - 2025 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+# frozen_string_literal: true
+
+class AddHmisUnitGroups < ActiveRecord::Migration[7.1]
+  def change
+    create_table :hmis_unit_groups do |t|
+      t.string :name, null: false
+      t.references :project, null: false, foreign_key: { to_table: :Project }
+      t.string :workflow_template_identifier, null: true # which workflow template to use to fill units in this group. use identifier instead of db id to avoid tying to a particular version of the template. nullable for non-CE usage
+      t.timestamps
+      t.timestamp :deleted_at
+    end
+
+    add_column :hmis_units, :hmis_unit_group_id, :integer, null: true
```

Comment:
to discuss: populating all existing units into groups, so that we don't need to deal with supporting "floating" units in the frontend. can be deferred to another pr

Reply:
I think we should use either `bigint` or `references` for pks, we're trying to get away from mismatched pk types.
```suggestion
    add_reference :hmis_units, :hmis_unit_group # adds index by default
```

Reply:
Originally I had `add_reference` but I was nervous about using `safety_assured` (or `, index: {algorithm: :concurrently}`) since this table is in use.  It seems like in similar recent migrations we just used `safety_assured` so I can switch back to that, and do more research onto whether its really safe